app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -11,6 +11,10 @@ class OmniauthCallbacksController < Devise::OmniauthCallbacksController
 
     def handle(provider)
       auth_hash = request.env['omniauth.auth']
+      if auth_hash.info.email.nil?
+        flash[:error] = "Email field is missing in your #{provider} account"
+        return redirect_to new_user_registration_path
```

Comment:
Why return?

Reply:
That was indeed by mistake.I'll correct it.

Reply:
Please do

Reply:
done

Reply:
@differentreality I realised that this "return" was intentionally written by me, since I was facing problem in redirecting when I removed "return" .
Please see if you could reproduce this error when removing return keyword

Reply:
it was not redirecting, instead gave me an error the following line:-
https://github.com/openSUSE/osem/blob/4785293bf1348549ade1c68ef35f613eaa509a6f/app/controllers/users/omniauth_callbacks_controller.rb#L14